src/Pages/GlobalConfigurations/Authorization/Shared/components/AppPermissions/RoleSelector.tsx

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (c) 2024. Devtron Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useMemo } from 'react'
+import { ACCESS_TYPE_MAP, SelectPickerOptionType } from '@devtron-labs/devtron-fe-common-lib'
+import Select, { components, MenuListProps, ValueContainerProps } from 'react-select'
+import { importComponentFromFELibrary } from '../../../../../../components/common'
+import { SELECT_ALL_VALUE } from '../../../../../../config'
+import { useAuthorizationContext } from '../../../AuthorizationProvider'
+import { CONFIG_APPROVER_ACTION, ARTIFACT_PROMOTER_ACTION, TERMINAL_EXEC_ACTION } from '../../../constants'
+import { roleSelectStyles } from './constants'
+import { getPrimaryRoleIndex, parseData } from '../../../utils'
+import { DirectPermissionRowProps } from './types'
+
+const ApproverPermission = importComponentFromFELibrary('ApproverPermission')
+
+const RoleSelector = ({
```

Comment:
This along with workflow selector, env selector and app/job selector have been abstracted from direct permission

Reply:
Also, role selector cannot be added using select picker at present